py/mattak/Dataset.py

```diff
@@ -423,3 +429,44 @@ def read_run_config(path : str) -> dict:
         conf = libconf.load(f)
 
     return conf
+
+# store for one station
+@lru_cache(maxsize=7)
```

Comment:
maxsize should be (for maximum performance) of the power of two, i.e., `maxsize= 2**n`. And I would be also more generous for more stations in the future. Use 64.